File: gaussian_basis/gaussian_basis/gaussian1d.py
import numpy as np


class Gaussian1D:
    r0: float
    ang: int
    orb_exp: float

    def __init__(self, orb_exp: float, r0: float, ang: int):
        self.orb_exp = orb_exp
        self.r0 = r0
        self.ang = ang

    def __call__(self, x: np.ndarray) -> np.ndarray:
        ang = self.ang
        r0 = self.r0
        orb_exp = self.orb_exp
        return (x - r0)**ang*np.exp(-orb_exp*(x - r0)**2)

    def __add__(self, n: int) -> 'Gaussian1D':
        position = self.position()
        orb_exp = self.orbital_exponent()
        angular = self.angular()
        return Gaussian1D(orb_exp=orb_exp, r0=position,
                          ang=angular + n)

    def __sub__(self, n: int) -> 'Gaussian1D':
        position = self.position()
        orb_exp = self.orbital_exponent()
        angular = self.angular()
        return Gaussian1D(orb_exp=orb_exp, r0=position,
                          ang=angular - n)

    def position(self) -> float:
        return self.r0

    def orbital_exponent(self) -> float:
        return self.orb_exp

    def angular(self) -> int:
        return self.ang

File: gaussian_basis/gaussian_basis/src2/integrals/gaussian1d.cpp
#include <cmath>
#include "gaussian1d.hpp"


Gaussian1D::Gaussian1D() {
    this->orb_exp = 0.0;
    this->r0 = 0.0;
    this->ang = 0;
}

Gaussian1D::Gaussian1D(double orb_exp, double r0, long ang) {
    this->orb_exp = orb_exp;
    this->r0 = r0;
    this->ang = ang;
}

double Gaussian1D::operator()(double x) const {
    return pow(x - r0, ang)*exp(-orb_exp*(x - r0)*(x - r0));
}

double Gaussian1D::position() const {
    return this->r0;
}
double Gaussian1D::orbital_exponent() const {
    return this->orb_exp;
}
long Gaussian1D::angular() const {
    return this->ang;
}

Gaussian1D Gaussian1D::operator+(int n) {
    return Gaussian1D(orb_exp, r0, ang + n);
}

Gaussian1D Gaussian1D::operator-(int n) {
    return Gaussian1D(orb_exp, r0, ang - n);
}
